### DISCUSSION TITLE
Possibility to monitor current requests

### DISCUSSION THREAD

**Discussion by dgodoyAtsistemas** [javamelody/javamelody#1226] [Q&A]
Hi guys,

We need to monitor the number of "Current Requests" that appears in JavaMelody to identify possible queues. **How does the app do it (requests, code, console commands...)?**

The main idea is to be able to control this number of current requests and establish thresholds in AWS that, when reached, send alerts to warn that the established request limit is being reached.

Thank you!

**Comment by evernat** [comment]
The number of "Current Requests" is the value in the "Active threads" graph of javamelody.

You can send metrics every minute, including the activeThreads metric, to an external service such as AWS CloudWatch.
There is a feature for that in javamelody, see https://github.com/javamelody/javamelody/wiki/UserGuideAdvanced#sending-metrics-to-aws-cloudwatch
After that you can set an alarm, see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchAlarms

Or if you prefer another service such as Prometheus, StatsD, Datadog or other see the currently available list at https://github.com/javamelody/javamelody/wiki/UserGuideAdvanced

If you prefer using a single http request, you can call the monitoring like http://youhost/yourwebapp/monitoring?part=lastValue&graph=activeThreads and get the value.

If you prefer to do by code, you can call javamelody public classes. You can get the metric value as much as you want using `net.bull.javamelody.JdbcWrapper.getActiveThreadCount()`.

**Reply by dgodoyAtsistemas** [reply]
Thanks evernat. Really appreciate it!
